// Tutorial.proto

// Enable generation of Protocol Buffers RPC services.
option cc_generic_services = true;
option java_generic_services = true;
option py_generic_services = true;

// Defining a search service.

// Define a request.

message SearchRequest {
  required string query = 1;
  optional int32 page_number = 2;
  optional int32 result_per_page = 3 [default = 10];
  enum Corpus {
    UNIVERSAL = 0;
    WEB = 1;
    IMAGES = 2;
    LOCAL = 3;
    NEWS = 4;
    PRODUCTS = 5;
    VIDEO = 6;
  }
  optional Corpus corpus = 4 [default = UNIVERSAL];
}

// Define a response

message SearchResponse {
  repeated group Result = 1 {
    required string url = 2;
    optional string title = 3;
    repeated string snippets = 4;
  }
}

// Define the service.

service SearchService 
{
  rpc Search (SearchRequest) returns (SearchResponse);
}

